### ISSUE TITLE
semantic: no error happens with `function a () {} var a = ''`

### ISSUE THREAD

**Issue by sapphi-red** [oxc-project/oxc#18719]
An error should be output for the following code, but semantic does not report any errors:
```js
function a() {}
var a = ''
```
[playground](https://playground.oxc.rs/?t=ast&code=function+a%28%29+%7B%7D%0Avar+a+%3D+%27%27%0A)
When I run this on Node, I get `SyntaxError: Identifier 'a' has already been declared`.

The error is properly shown for
```js
var a = ''
function a() {}
```

Maybe related to #15961
